storage/temptable/include/temptable/allocator.h

```diff
@@ -35,6 +35,8 @@ TempTable custom allocator. */
 
 #include "my_dbug.h"
 #include "my_sys.h"
+#include "mysql/components/services/log_builtins.h"
```

Comment:
@dutow my concern here is that surprisingly, for some reason, logging infrastructure is not used inside the `temptable` code. And I am hesitating to introduce this dependency. Something tells me that there are hidden reasons for that of which we are unaware.
So, although it was me who asked you to add a new error message, I am leaning towards completely removing it. Sorry for that.
However, even if we decide to leave it, these two headers are pretty heavy and should be moved to the `allocator..cc` file along with the `Prefer_RAM_over_MMAP_policy_obeying_per_table_limit::Source block_source()` method definition.

Reply:
We could go back to my initial suggestion, adding a warning message to the system variable instead.

Reply:
Yes, let's do this